qds_sdk/sensors.py

```diff
@@ -0,0 +1,71 @@
+"""
+The sensors module contains the base definition for a generic
+sensor call and the implementation of all the specific sensors
+"""
+
+from __future__ import print_function
+from qds_sdk.qubole import Qubole
+from qds_sdk.resource import Resource
+from argparse import ArgumentParser
+
+import logging
+import json
+
+log = logging.getLogger("qds_sensors")
+
+
+class Sensor(Resource):
+    """
+    qds_sdk.Sensor is the base Qubole sensor class. Different types of Qubole
+    sensors can subclass this.
+    """
+
+    @classmethod
+    def check(cls, data):
+        """
+        Method to call the sensors api with json payload
+        :param data: valid json object
+        :return: True or False
+        """
+        conn = Qubole.agent()
+        return conn.post(cls.rest_entity_path, data=data)['status']
+
+    @classmethod
+    def check_cli(cls, args):
+        """
+        Method to call check after parsing args from cli
+        :param args: inline arguments
+        :return: True or False
+        """
+        parser = cls.parsers()
+        parsed = parser.parse_args(args)
+        return cls.check(json.loads(parsed.data))
+
+    @classmethod
+    def parsers(cls):
```

Comment:
Any specific reason to move these methods into new `SensorCmdLine` class? I followed the design of `Command` class here.. Also I think `cls` variable won't be available in parent class, as its not inside a class method?